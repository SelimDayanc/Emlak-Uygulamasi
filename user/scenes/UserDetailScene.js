/**
 @flow
 */
import React, {Component, PropTypes} from 'react';
import {
  Dimensions,
  Image,
  StyleSheet,
  Text,
  TouchableHighlight,
  View,
} from 'react-native';
import colors from '../../common/colors';
import FontAwesome from 'react-native-vector-icons/FontAwesome';
import moment from 'moment';

export default class UserDetailScene extends Component {
  static propTypes = {
    user: PropTypes.object.isRequired,
  };

  render() {
    const {user, loadScene} = this.props;

    return (
      <View style={styles.container}>
        {user.image
          ? <Image
              source={{uri: user.image}}
              style={styles.logo}
              resizeMode="cover"
            />
          : <FontAwesome
              name="picture-o"
              color="white"
              size={200}
              style={styles.emptyImageIcon}
            />}

        <View style={styles.editIconWrapper}>
          <TouchableHighlight onPress={loadScene} underlayColor="transparent">
            <FontAwesome
              name="pencil"
              color={colors.darkGrey}
              size={18}
              style={styles.editIcon}
            />
          </TouchableHighlight>
        </View>

        <View style={styles.content}>
          <Text style={styles.username}>{user.name}</Text>
          <Text style={styles.date}>
            Member Since {moment(user.created_at).format('MMMM D YYYY')}
          </Text>
        </View>

      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'white',
  },
  content: {
    padding: 20,
  },
  username: {
    fontSize: 20,
    fontWeight: '700',
    color: colors.darkGrey,
  },
  date: {
    paddingVertical: 10,
    fontSize: 15,
    fontWeight: '100',
    color: colors.smokeGreyDark,
  },
  logo: {
    height: 200,
    width: Dimensions.get('window').width,
  },
  emptyImageIcon: {
    height: 200,
    backgroundColor: colors.smokeGreyLight,
    textAlign: 'center',
  },
  editIconWrapper: {
    position: 'absolute',
    top: 200,
    right: 15,
    marginTop: -20,
    height: 40,
    width: 40,
    borderRadius: 20,
    zIndex: 1000,
    alignItems: 'center',
    justifyContent: 'center',
    shadowColor: colors.smokeGreyDark,
    shadowOpacity: 0.6,
    shadowOffset: {width: 1, height: 1},
  },
});
