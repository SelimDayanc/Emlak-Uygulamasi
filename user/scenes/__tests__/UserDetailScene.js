import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import UserDetailScene from '../UserDetailScene';

test('renders heart', () => {
  const tree = renderer
    .create(<UserDetailScene user={{name: 'zal', email: 'z4ls@live.com'}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
