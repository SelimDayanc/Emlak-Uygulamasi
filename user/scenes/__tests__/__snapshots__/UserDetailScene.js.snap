// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders heart 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "white",
          "fontSize": 200,
        },
        Object {
          "backgroundColor": "#e3e7ed",
          "height": 200,
          "textAlign": "center",
        },
        Object {
          "fontFamily": "FontAwesome",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 20,
        "height": 40,
        "justifyContent": "center",
        "marginTop": -20,
        "position": "absolute",
        "right": 15,
        "shadowColor": "#5d5f63",
        "shadowOffset": Object {
          "height": 1,
          "width": 1,
        },
        "shadowOpacity": 0.6,
        "top": 200,
        "width": 40,
        "zIndex": 1000,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#484848",
              "fontSize": 18,
            },
            undefined,
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "padding": 20,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#484848",
          "fontSize": 20,
          "fontWeight": "700",
        }
      }
    >
      zal
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#5d5f63",
          "fontSize": 15,
          "fontWeight": "100",
          "paddingVertical": 10,
        }
      }
    >
      Member Since May 2015
    </Text>
  </View>
</View>
`;
